src/System.Data.SqlClient/src/System/Data/ProviderBase/DbConnectionPoolIdentity.Windows.cs

```diff
@@ -22,8 +22,9 @@ private static DbConnectionPoolIdentity GetCurrentNative()
             using (WindowsIdentity identity = WindowsIdentity.GetCurrent())
             {
                 IntPtr token = identity.AccessToken.DangerousGetHandle();
-                bool isNetwork = identity.User.IsWellKnown(WellKnownSidType.NetworkSid);
-                string sidString = identity.User.Value;
+                SecurityIdentifier user = identity.User;
+                bool isNetwork = user.IsWellKnown(WellKnownSidType.NetworkSid);
+                string sidString = user.ToString();
```

Comment:
Why use ToString()? The docs mention that Value provides the upper case value of SID and ToString() has no such guarantees. 
It makes sense to leave this part as is.

Reply:
For the sake on consistency, I think it makes more sense to use the `Value` instead of `ToString()`

Reply:
I rewrote the code for ToString() to be faster in https://github.com/dotnet/corefx/pull/33579 In fact it is guaranteed that the ToString() return value has an uppercase S in it because it's a literal. 

Calling ToUpperCase() on it is a waste of time. There is no need for the returned value to be uppercased only that it be consistent with every call, the only use is as part of the pool key so case is irrelevant. I judged it better not to waste time making a call that can be proven to have no effect.

Reply:
The part that is concerning to me is that knowing that ToString() implementation has an uppercase S is an internal detail vs there is an API called `.Value` which guarantees upper case string. 

In terms of API consistency, if this code is to execute on another framework which doesn't have the same guarantees of internals, that you are describing, then it may fall apart. 

You improved the performance of ToString(), which means that there should be an overall improvement with .Value retrieval as well on .Net Core

Reply:
Done, the perf change isn't worth worrying about.